spec/support/faux_writer.rb

```diff
@@ -1,11 +1,28 @@
+require 'net/http'
+
 require 'datadog/tracing/writer'
 
 require 'support/faux_transport'
+require 'support/network_helpers'
 
 # FauxWriter is a dummy writer that buffers spans locally.
 class FauxWriter < Datadog::Tracing::Writer
+  include NetworkHelpers
+
   def initialize(options = {})
-    options[:transport] ||= FauxTransport.new
+    options[:transport] ||= if test_agent_running?
+                              Datadog::Transport::HTTP.default(
+                                headers: {
+                                  'X-Datadog-Trace-Env-Variables' => ENV.to_h.select { |key, _| key.start_with?('DD_') }
```

Comment:
Is `X-Datadog-Trace-Env-Variables` supposed to be an official tracer feature that `dd-trace-rb` does not support yet?

Reply:
This is not an official tracer feature. The purpose of this is to header is to include the full tracer configuration, which is later used in the Test Agent for assertions (like service-naming by checking the Schema Version used).

Reply:
@marcotc Though on second thought, to achieve this, I believe that I need to include the tracer settings used as well since they can override / be used in addition to the configurable environment variables, correct?

Reply:
Hmmm, do we know what tracer configuration we are looking for?

It is correct that `ENV` won't give you the tracer configuration.

I ask because we have a lot of tracer configuration options (I'd guess 100+). We probably don't need them all.

The effective configuration used by the tracer is in the `Datadog.configuration` object (it has a `.to_h` to give you all options, for example)

Reply:
Mainly the configuration that we would like to send over is basic datadog stuff like DD_SERVICE, the span attribute schema version being used (DD_TRACE_SPAN_ATTRIBUTE_SCHEMA), and a few more service naming specific variables.

Reply:
Ok, the best way here is to collect the require data individually from `Datadog.configure` (e.g. `Datadog.configure.service`) and send each one as a header.

Reply:
great, I made this change!